lib/plugins/aws/deploy/lib/uploadArtifacts.test.js

```diff
@@ -193,6 +199,30 @@ describe('uploadArtifacts', () => {
       });
     });
 
+    it('should not upload the .zip file to the S3 bucket when hash is identical', () => {
+      cryptoStub
+        .createHash()
+        .update()
+        .digest.onCall(0)
+        .returns('local-hash-zip-file');
+
+      uploadStub.onCall(0).resolves({
+        Metadata: {
+          filesha256: 'local-hash-zip-file',
+        },
+      });
+
+      const tmpDirPath = getTmpDirPath();
+      const artifactFilePath = path.join(tmpDirPath, 'artifact.zip');
+      serverless.utils.writeFileSync(artifactFilePath, 'artifact.zip file content');
+
+      return awsDeploy.uploadZipFile(artifactFilePath).then(() => {
+        expect(uploadStub).to.have.been.calledOnce;
```

Comment:
Note to reviewers) `uploadStub` should be called once (called for `headObject`, not `upload`).